pkg/apis/triggers/v1alpha1/trigger_defaults.go

```diff
@@ -58,3 +59,11 @@ func (t triggerSpecBindingArray) defaultBindings() {
 		}
 	}
 }
+
+func templateNameToRef(template *TriggerSpecTemplate) {
+	name := template.Name
+	if name != "" && template.Ref == nil {
+		template.Ref = &name
+		template.Name = ""
```

Comment:
just a thought is it required to set "" to `name` can't we ignore it ?

Reply:
defaults get applied before validation, if we do not unset Name (to ""), we'll hit this line in trigger_validation.go:

```
      if t.Name != "" && t.Ref != nil {
		errs = errs.Also(apis.ErrMultipleOneOf("template.name", "template.ref"))
	}
```

Reply:
ah i see got it :+1: